python/fledge/services/core/api/statistics.py

```diff
@@ -148,3 +149,73 @@ async def get_statistics_history(request):
+
+      Implementation:
+          Calculation via: (sum(value) / count(value)) * 60 / (<statistic history interval>)
+          Queries for above example:
+          select key, 4 * (sum(value) / count(value)) from statistics_history where history_ts >= datetime('now', '-1 Minute') and key in ("SINUSOID", "FASTSINUSOID", "READINGS" ) group by key;
+          select key, 4 * (sum(value) / count(value)) from statistics_history where history_ts >= datetime('now', '-5 Minute') and key in ("SINUSOID", "FASTSINUSOID", "READINGS" ) group by key;
+          select key, 4 * (sum(value) / count(value)) from statistics_history where history_ts >= datetime('now', '-15 Minute') and key in ("SINUSOID", "FASTSINUSOID", "READINGS" ) group by key;
+      """
+    params = request.query
+    if 'periods' not in params:
+        raise web.HTTPBadRequest(reason="periods request parameter is required")
+    if 'statistics' not in params:
+        raise web.HTTPBadRequest(reason="statistics request parameter is required")
+
+    if params['periods'] == '':
+        raise web.HTTPBadRequest(reason="periods cannot be an empty. Also comma separated list of values required "
+                                        "in case of multiple periods of time")
+    if params['statistics'] == '':
+        raise web.HTTPBadRequest(reason="statistics cannot be an empty. Also comma separated list of statistics values "
+                                        "required in case of multiple assets")
+
+    periods = params['periods']
+    period_split_list = list(filter(None, periods.split(',')))
+    if not all(p.isdigit() for p in period_split_list):
+        raise web.HTTPBadRequest(reason="periods should contain numbers")
+    # 1 week = 10080 mins
+    if any(int(p) > 10800 for p in period_split_list):
+        raise web.HTTPBadRequest(reason="The maximum allowed value for a period is 10080 minutes")
+
+    stats = params['statistics']
+    stat_split_list = list(filter(None, [x.upper() for x in stats.split(',')]))
+    storage_client = connect.get_storage_async()
+    # To find the interval in secs from stats collector schedule
+    scheduler_payload = PayloadBuilder().SELECT("schedule_interval").WHERE(
+        ['process_name', '=', 'stats collector']).payload()
+    result = await storage_client.query_tbl_with_payload('schedules', scheduler_payload)
+    if len(result['rows']) > 0:
+        scheduler = Scheduler()
+        interval_days, interval_dt = scheduler.extract_day_time_from_interval(result['rows'][0]['schedule_interval'])
+        interval = datetime.timedelta(days=interval_days, hours=interval_dt.hour, minutes=interval_dt.minute,
+                                      seconds=interval_dt.second)
+        interval_in_secs = interval.total_seconds()
+    else:
+        raise web.HTTPNotFound(reason="No stats collector schedule found")
+    formula_str = "(sum(value) / count(value)) * 60 / {}".format(int(interval_in_secs))
```

Comment:
It would be better to use the aggregate stuff in the Storage API to get the count and sum and then do the calculation in the python code. .This would be more portable sand would probably solve your Postgres issue.